File: src/robot/robot.jl
module Robot
include("sensor.jl")
include("robot-components.jl")
import Loc.Location
import Loc.MoveDirection

struct State
	location::Location
	moveDirection::MoveDirection
	velocity:Float16 = 1

	State(loc::Location, md::MoveDirection) = new(loc, md)
	State(loc::Location, md::MoveDirection, vel::Float16) = new(loc, md, vel)
end


mutable struct GenerativeRobot
	# sensors::Sensor
	currentState::State
	
	# rightRearWheel::Wheel
	# leftRearWheel::Wheel
	# rightFrontWheel::Wheel
	# leftFrontWheel::Wheel
end

function move(robot::GenerativeRobot, direction::MoveDirection, location::Location) 
	s = State(location, direction)
	robot.currentState = s
	data = Dict(
		"topic" => Topics[:position],
		"data" => s
	)
	publish(Topics[:position], data)
end

export GenerativeRobot, State
end

File: py/robot/robot.py
# import RobotComponents as rc
from env.location import Location
from env.location import MoveDirection

class State:
	def __init__(self, location, moveDirection, velocity) -> None:
		self.location = location
		self.moveDirection = moveDirection
		self.velocity = velocity

	def __str__(self) -> str:
		return "Location: " + self.location + " MD: " + self.moveDirection + " vel: " + self.velocity


class GenerativeRobot:
	def __init__(self, state) -> None:
		self.currentState = state

	def __str__(self) -> str:
		return "state: " + self.currentState
	
	def move(self, direction, location):
		s = State(location, direction)
		self.currentState = s
		# TODO send pubsub